### PR TITLE
Add pass-through query documentation

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 1]
```diff
@@ -262,6 +262,25 @@ as part of the table name, separated by a colon. For example:
 .. _full text query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
 
 
+Pass-through Queries
+--------------------
+
+The Elasticsearch connector allows you to embed any valid Elasticsearch query,
+that uses the `ES Query DLS
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:269]: DSL. Also, I wouldn't abbreviate ES

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 2]
```diff
@@ -262,6 +262,25 @@ as part of the table name, separated by a colon. For example:
...
+The results can then be used in any SQL statement, wrapping the ES query. The
+syntax extends the syntax of the enhanced ES table names with the following::
+
+    SELECT * FROM es.default."<index>$query:<es-query>"
+
+The ES query string ``es-query`` is base32-encoded to avoid having to deal with
+escaping quotes and case sensitivity issues in table identifiers.
+
+The result of these query tables is a table with a single row and a single
+column named ``result`` of type JSON.
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:282]: The result is VARCHAR, not JSON, but it contains the JSON payload returned by Elasticsearch, and can be processed with the built-in JSON functions in presto (https://prestosql.io/docs/current/functions/json.html)

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/elasticsearch.rst]: Updated

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst (hunk 1 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/elasticsearch.rst]: Done

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 3]
```diff
@@ -262,6 +262,27 @@ as part of the table name, separated by a colon. For example:
...
 
+Pass-through Queries
+--------------------
+
+The Elasticsearch connector allows you to embed any valid Elasticsearch query,
+that uses the `Elasticsearch Query DLS
+<https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html>`_
+in your SQL query.
+
+The results can then be used in any SQL statement, wrapping the ES query. The
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:273]: "the Elasticsearch query"

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 4]
```diff
@@ -262,6 +262,27 @@ as part of the table name, separated by a colon. For example:
 .. _full text query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
 
 
+Pass-through Queries
+--------------------
+
+The Elasticsearch connector allows you to embed any valid Elasticsearch query,
+that uses the `Elasticsearch Query DLS
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: DSL

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 5]
```diff
@@ -262,6 +262,27 @@ as part of the table name, separated by a colon. For example:
...
+<https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html>`_
+in your SQL query.
+
+The results can then be used in any SQL statement, wrapping the ES query. The
+syntax extends the syntax of the enhanced ES table names with the following::
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:274]: ES->Elasticsearch

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 6]
```diff
@@ -262,6 +262,27 @@ as part of the table name, separated by a colon. For example:
...
+
+    SELECT * FROM es.default."<index>$query:<es-query>"
+
+The ES query string ``es-query`` is base32-encoded to avoid having to deal with
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:278]: ES->Elasticsearch

**Comment by martint** [comment]
@mosabua, there's a build failure:

```
Warning, treated as error:
/Users/martin/projects/presto/presto/presto-docs/src/main/sphinx/connector/elasticsearch.rst:282:undefined label: /functions/json (if the link has no caption the label must precede a section header)
make: *** [html] Error 2
```

**Comment by mosabua** [comment]
> @mosabua, there's a build failure:
> 
> ```
> Warning, treated as error:
> /Users/martin/projects/presto/presto/presto-docs/src/main/sphinx/connector/elasticsearch.rst:282:undefined label: /functions/json (if the link has no caption the label must precede a section header)
> make: *** [html] Error 2
> ```

Sorry ... wrong link. ref instead of doc.. will push fix in a sec

**Comment by martint** [comment]
Build failure is unrelated

**Pull request merged**